dbt_subprojects/hourly_spellbook/models/_project/balancer/pools/balancer_view_pools_latest.sql

```diff
@@ -0,0 +1,37 @@
+{{ config(
+        schema = 'balancer',
+        alias = 'view_pools_latest', 
+        post_hook='{{ expose_spells(blockchains = \'["arbitrum", "avalanche_c", "base", "ethereum", "gnosis", "optimism", "polygon", "zkevm"]\',
+                                spell_type = "project",
+                                spell_name = "balancer",
+                                contributors = \'["viniabussafi"]\') }}'
+        )
+}}
+
+SELECT
+    m.blockchain,
+    m.pool_id,
+    m.pool_address,
+    m.pool_symbol,
+    m.pool_type,
+    m.factory_version,
+    m.factory_address,
+    m.creation_date,
+    SUM(CASE WHEN block_date = (SELECT MAX(block_date) FROM {{ref('balancer_pools_metrics_daily')}})
+    THEN tvl_usd END) AS tvl_usd,
+    SUM(CASE WHEN block_date = (SELECT MAX(block_date) FROM {{ref('balancer_pools_metrics_daily')}})
+    THEN tvl_eth END) AS tvl_eth,
+    SUM(swap_amount_usd) AS swap_volume,
+    SUM(CASE WHEN block_date > CURRENT_DATE - INTERVAL '365' day THEN swap_amount_usd END) AS swap_volume_1y,
+    SUM(CASE WHEN block_date > CURRENT_DATE - INTERVAL '30' day THEN swap_amount_usd END) AS swap_volume_30d,
+    SUM(CASE WHEN block_date > CURRENT_DATE - INTERVAL '7' day THEN swap_amount_usd END) AS swap_volume_7d,
+    SUM(CASE WHEN block_date > CURRENT_DATE - INTERVAL '1' day THEN swap_amount_usd END) AS swap_volume_1d,  
+    SUM(fee_amount_usd) AS fees_collected,
+    SUM(CASE WHEN block_date > CURRENT_DATE - INTERVAL '365' day THEN fee_amount_usd END) AS fees_collected_1y,
+    SUM(CASE WHEN block_date > CURRENT_DATE - INTERVAL '30' day THEN fee_amount_usd END) AS fees_collected_30d,
+    SUM(CASE WHEN block_date > CURRENT_DATE - INTERVAL '7' day THEN fee_amount_usd END) AS fees_collected_7d,
+    SUM(CASE WHEN block_date > CURRENT_DATE - INTERVAL '1' day THEN fee_amount_usd END) AS fees_collected_1d  
+FROM {{source('balancer', 'dataset_factory_pool_mapping')}} m
```

Comment:
```suggestion
FROM {{ source("balancer", "dataset_factory_pool_mapping", database="dune") }} m
```
we need to include a new database param to read these types of tables. we are extra cautious with these introduced into spellbook, as changes outside of the repo can then break spells like this reliant upon the table. what is this table exactly? how is it built, populated and maintained?

Reply:
Thanks for the information!

I created this [query](https://dune.com/queries/4080393), which is an enhanced version of our current pool labels (e.g., spellbook/dbt_subprojects/dex/models/_projects/balancer/labels/arbitrum/labels_balancer_v2_pools_arbitrum.sql). It not only labels the pools but also includes a mapping of pool factories by joining with another dataset (dune.balancer.dataset_balancer_deployments). This dataset is maintained by the Balancer Maxis team.

The query I provided would serve as the source for dune.balancer.dataset_factory_pool_mapping.

I think it would be more straightforward to manage this on our end by using the upload to build the view in this PR, but I’m open to any approach that works best for you.

Reply:
we can give it a shot as a spell view.
how much do you plan to build on top?
would it be balancer only downstream of this, not community wide spells?
are you comfortable with this + downstream spells built on top being excluded from prod as soon as a failure happens (if it does)?

Reply:
I don't plan to dig further down than as-is. Only change I can see right now would be also including balancer V3 pools once they start rolling in over the next few months, but keeping it in balancer's namespace as I think this is very specific to us.
I check dune.balancer.dataset_balancer_deployments at least weekly for almost an year, so I'll also be on the lookout for any errors/changes, as well.

Really appreciate your help!

Reply:
i'm not sure i understand the question?

Reply:
ah, yes, you need to apply my suggested change above where you add database param into source jinja macro. instead of `dune`, you would need to add your team name the mat view was built with. when you query this table on dune app, isn't it something like:
`<database_name>.<schema>.<alias>`?

Reply:
Thanks a lot and sorry for that! should be good now